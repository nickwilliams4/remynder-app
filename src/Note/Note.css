
@media screen and (max-width: 500px) {
ul {
  padding: 0;
}
}

.Note_edit, .Note__delete {
  height: 50px;
  width: 120px;
  border-radius: 20px;
  font-size: 16px;
  background-color: black;
  color: white;
  margin: 8px;
}

.button-icon {
  margin-left: 10px;
}

.reminder-action-button {
  cursor: pointer;
}

.Note {
  text-align: center;
}

.Note__title a {
  color: black;
}

.Note_edit:hover, .Note__delete:hover {
  color: black;
  background-color: white;
}


