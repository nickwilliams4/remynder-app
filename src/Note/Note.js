import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import ApiContext from '../ApiContext'
import config from '../config'
import moment from 'moment'
import './Note.css'
import TokenService from '../services/token-service'

export default class Note extends Component {
  static defaultProps ={
    onDeleteNote: () => {},
  }
  static contextType = ApiContext;

  handleClickDelete = e => {
    e.preventDefault()
    const note_id = this.props.id

    fetch(`${config.API_ENDPOINT}/notes/${note_id}`, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        'authorization': `bearer ${TokenService.getAuthToken()}`
      },
    })
      .then(res => {
        if (!res.ok)
          return Promise.reject('Could not delete')
      })
      .then(() => {
        this.context.deleteNote(note_id)
        this.props.onDeleteNote(note_id)
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render() {
    const { title, id, created, content, remynder } = this.props
    return (
      <div className='Note'>
      <h2 className='Note__title'>
        <Link to={`/note/${id}`}>
          {title}
        </Link>
      </h2>
      <h4>{content}</h4>
      <div className='Note__dates'>
        <div className='Note__dates-modified'>
          Created: 
          {' '}
          <span className='Date'>
            {moment(created).format('MMM Do YYYY')}
          </span>
        </div>
      </div>
      <div className='Note__Content'>
      </div>
      <p>Remynder: Every {remynder} hours</p>
      <Link to={`edit/${id}`}>
      <button
        className='Note_edit'
        type='button'
        >
          {''}
          Edit
        </button>
        </Link>
      <button
        className='Note__delete'
        type='button'
        onClick={this.handleClickDelete}
      >
        {' '}
        Delete
      </button>
    </div>
    )
  }
}

