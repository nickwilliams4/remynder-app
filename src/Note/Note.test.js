import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Note from './Note'

describe(`Note component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
  }

  it('renders the Note', () => {
    const wrapper = shallow(<Note {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
