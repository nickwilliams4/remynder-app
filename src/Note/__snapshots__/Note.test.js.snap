// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component renders the Note 1`] = `
<div
  className="Note"
>
  <h2
    className="Note__title"
  >
    <Link
      to="/note/a"
    />
  </h2>
  <h4 />
  <div
    className="Note__dates"
  >
    <div
      className="Note__dates-modified"
    >
      Created:
       
      <span
        className="Date"
      >
        Apr 21st 2020
      </span>
    </div>
  </div>
  <div
    className="Note__Content"
  />
  <p>
    Remynder: Every 
     hours
  </p>
  <Link
    to="edit/a"
  >
    <button
      className="Note_edit reminder-action-button"
      type="button"
    >
      Edit
      <FontAwesomeIcon
        border={false}
        className="button-icon"
        fixedWidth={false}
        flip={null}
        icon="edit"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </button>
  </Link>
  <button
    className="Note__delete reminder-action-button"
    onClick={[Function]}
    type="button"
  >
     
    Delete
    <FontAwesomeIcon
      border={false}
      className="button-icon"
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </button>
</div>
`;
